app/assets/javascripts/controllers/catalog/catalog_item_form_controller.js

```diff
@@ -403,5 +403,41 @@ ManageIQ.angular.app.controller('catalogItemFormController', ['$scope', 'catalog
     });
   }
 
+  $scope.cancelCopyProvisioning = function() {
+    closeConfirmationModal();
+  }
+
+  $scope.copyProvisioning = function() {
+    closeConfirmationModal();
+    vm.formOptions();
+    vm.catalogItemModel.retirement_repository_id = vm.catalogItemModel.provisioning_repository_id;
+    vm.catalogItemModel.retirement_playbook_id = vm.catalogItemModel.provisioning_playbook_id;
+    vm.catalogItemModel.retirement_machine_credential_id = vm.catalogItemModel.provisioning_machine_credential_id;
+    vm.catalogItemModel.retirement_network_credential_id = vm.catalogItemModel.provisioning_network_credential_id;
+    vm.catalogItemModel.retirement_cloud_credential_id = vm.catalogItemModel.provisioning_cloud_credential_id;
+    vm.catalogItemModel.retirement_inventory = vm.catalogItemModel.provisioning_inventory;
+    vm.catalogItemModel.retirement_dialog_existing = vm.catalogItemModel.provisioning_dialog_existing;
+    vm.catalogItemModel.retirement_dialog_id = vm.catalogItemModel.provisioning_dialog_id;
+    vm.catalogItemModel.retirement_dialog_name = vm.catalogItemModel.provisioning_dialog_name;
+    vm.catalogItemModel.retirement_key = '';
+    vm.catalogItemModel.retirement_value = '';
+    vm.catalogItemModel.retirement_variables = angular.copy(vm.catalogItemModel.provisioning_variables);
```

Comment:
@syncrou that is already set when the form loads, no need to set when copying provisioning fields.